cmd/promtool/main.go

```diff
@@ -421,6 +431,40 @@ func CheckMetrics() int {
 	return 0
 }
 
+// PrettifyRules prettifies the rule files.
+func PrettifyRules(files ...string) int {
+	var errors []error
+	for _, filePath := range files {
+		rgs, errs := rulefmt.ParseFile(filePath)
+		if len(errs) > 0 {
+			errors = append(errors, errs...)
+			continue
+		}
+		for i, groups := range rgs.Groups {
+			for j, rules := range groups.Rules {
+				prettifiedExpr, err := prettier.New(rules.Expr.Value).Prettify()
+				if err != nil {
+					errors = append(errors, err)
+					continue
+				}
+				rgs.Groups[i].Rules[j].Expr.SetString(prettifiedExpr)
+			}
+		}
+		data, err := yaml.Marshal(rgs)
+		if err != nil {
+			panic(err)
+		}
+		if err = ioutil.WriteFile(filePath, data, 0666); err != nil {
+			fmt.Fprintln(os.Stderr, fmt.Sprintf("failed formatted %s file:", filePath), err.Error())
```

Comment:
Shouldn't this return a failure?

Reply:
Currently, the reason why we are not stopping at that point is, we want to write those files that are perfect to do so and print errors for the ones that cause any error.

Should we stop the moment we encounter error while writing any file, without caring about the remaining ones?

Reply:
For I/O errors we should probably stop.

Reply:
Also this way of writing files is not atomic, so a failure could lead to data loss.

Reply:
Addressed this. Now, this will stop and `return 1` in case of an error.

Reply:
The writing of the files is still not atomic.

Reply:
Yeah, consider what happens if you run out of disk space half way through writing the file.

Reply:
Writing rules file now done using temp file and then renaming it. I have, that addresses our atomic issue.